### PR TITLE
chore: update prettier and eslint-config-prettier in preparation of upgrading TypeScript to v5

### PR DESCRIPTION
**Pull request by chrisdholt**
# Pull Request

## 📖 Description
This PR updates prettier and eslint-config-prettier dependencies to prep or an update to TypeScript v5. The first commit is the updating of dependencies. The second commit is the updated files. The third commit is change files.

### 🎫 Issues
Closes #6359

### CODE REVIEW

File: packages/web-components/fast-ssr/src/styles/fast-style.fixture.html [hunk 1]
```diff
@@ -388,11 +388,11 @@ <h2>Heading</h2>
                 </div>
             </template>
         </fast-card>
-        <fast-card style="width: 300px;">
+        <fast-card style="width: 300px">
             <template shadowroot="open">
                 <fast-style style-id="fast-card"></fast-style>
                 <img src="https://via.placeholder.com/300x200/414141" />
-                <div style="padding: 0 10px 10px; color: var(--neutral-foreground-rest);">
+                <div style="padding: 0 10px 10px; color: var(--neutral-foreground-rest)">
```
Comment (scomea) [inline packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: Are we sure about not having the trailing semi-colon?

Comment (bheston) [reply packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: I support the trailing semicolon as it's how I personally write manual css and it aligns with other language decisions like maintaining a trailing comma or even the semicolon in JS which isn't always strictly required. Is there an option to turn it on?

Comment (chrisdholt) [reply packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: We are opted into the only configuration for semi colons. This is prescriptive on Prettier's part and not something we can adjust with current configuration. On the plus side, you can continue to use these and prettier will take care of removing as necessary. If they change to include it, we can update and it'll include it.

Comment (bheston) [reply packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: Hmm, maybe the trailing comma is gone, at least it wasn't added in some places I would have expected it. I don't have a strong opinion on this though (aside from still not agreeing with Prettier's model in general).

**Review by bheston — APPROVED** [review]

**Review by scomea — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-ssr/src/styles/fast-style.fixture.html (hunk 1 above)
Comment (Falkicon) [reply packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: My style has generally been to not include the trailing SC, but I don't think there is a right or wrong here so whatever people want is cool with me.

Comment (chrisdholt) [reply packages/web-components/fast-ssr/src/styles/fast-style.fixture.html:395]: We don’t have a choice so folks can leave it in or remove but prettier is gonna do what it wants :)

**Review by awentzel — APPROVED** [review]

**Pull request merged**